src/Mutation/Mutation.php

```diff
@@ -42,9 +42,11 @@
 use PhpParser\Node;
 
 /**
+ * @final
+ *
  * @internal
  */
-final class Mutation implements MutationInterface
+class Mutation
```

Comment:
I tend to think that having an interface and a closed class just by its own virtue has a merit, even if don't plan to export that interface. I might be spoiled by the fruit of Go, where you can't make a step without an interface of some sorts, but that's the better way because you naturally avoid all the problems of inheritance.

Reply:
I mean the class would be final (`@final` only for now though) so inheritance is definitely out of question here.

The question is more that `Mutation` is more an object for the _engine_ to consume: it is not where you want to allow an extension point because that would mean allowing the engine to be extended (I don't mean by inheritance but in terms of features/behaviour) in some ways which is a very different idea

Reply:
I see that `MutationInterface` is tagged `@internal` so it isn't nor wasn't an extension point per se.